lightning/src/ln/features.rs

```diff
@@ -242,6 +244,18 @@ mod sealed {
 				}
 			}
 
+			impl <T: $feature> Features<T> {
+				/// Set this feature as supported.
+				pub fn $supports_setter(&mut self) {
+					<T as $feature>::set_optional_bit(&mut self.flags);
+				}
+
+				/// Set this feature as required.
+				pub fn $requires_setter(&mut self) {
+					<T as $feature>::set_required_bit(&mut self.flags);
+				}
```

Comment:
Could you use the fluent-style signature (i.e., consume `mut self` rather than take `&mut self`) as is done in `clear_upfront_shutdown_script`? This allows for stringing these together.

Reply:
@TheBlueMatt Is this fine for the bindings, or do you wanna put this off for now?

Reply:
I'd say its fine, it'll result in a bunch of needless `clone`s in Java, but hopefully if we take invoice upstream it'll be rare that people need to use it much anyway.